gateway/oauth_manager.go

```diff
@@ -688,10 +712,28 @@ func (r *RedisOsinStorageInterface) SetClient(id string, client osin.Client, ign
 
 	r.store.SetKey(key, string(clientDataJSON), 0)
 
-	log.Debug("Storing copy in set")
-
 	keyForSet := prefixClientset + prefixClient // Org ID
 
+	log.Debug("Storing key in oauth clients index list")
+	indexKey := prefixClientIndexList + orgID
+	//check if the indexKey exists
+	exists, err := r.store.Exists(indexKey)
+	if err != nil {
+		return err
+	}
+	// if it exists, delete the key to avoid duplicity in the client index list
+	if exists {
+		keys, _ := r.store.GetListRange(indexKey, 0, -1)
```

Comment:
Can we instead of retrieving the list all the time, we can just do the "lrem" of this value?

Reply:
solved